core/src/main/java/smile/classification/OneVersusRest.java

```diff
@@ -138,12 +150,58 @@ public OneVersusRest(Classifier<T>[] classifiers, PlattScaling[] platts, IntSet
         return new OneVersusRest<>(classifiers, platts);
     }
 
+    /**
+     * Fits a multi-class model with binary classifiers.
+     * Use +1 and -1 as positive and negative class labels.
+     * @param formula a symbolic description of the model to be fitted.
+     * @param data the data frame of the explanatory and response variables.
+     * @param trainer the lambda to train binary classifiers.
+     */
+    public static OneVersusRest<Tuple> fit(Formula formula, DataFrame data, BiFunction<Formula, DataFrame, SoftClassifier<Tuple>> trainer) {
+        return fit(formula, data, +1, -1, trainer);
+    }
+
+    /**
+     * Fits a multi-class model with binary classifiers.
+     * Use +1 and -1 as positive and negative class labels.
+     * @param formula a symbolic description of the model to be fitted.
+     * @param data the data frame of the explanatory and response variables.
+     * @param pos the class label for one case.
+     * @param neg the class label for rest cases.
+     * @param trainer the lambda to train binary classifiers.
+     */
+    public static OneVersusRest<Tuple> fit(Formula formula, DataFrame data, int pos, int neg, BiFunction<Formula, DataFrame, SoftClassifier<Tuple>> trainer) {
+        formula = formula.expand(data.schema());
+        DataFrame x = formula.x(data);
+        BaseVector bv = formula.y(data);
+
+        ClassLabels codec = ClassLabels.fit(bv);
+        int k = codec.k;
+        if (k <= 2) {
+            throw new IllegalArgumentException("Only %d classes" + k);
+        }
+
+        int n = x.nrows();
+        int[] y = codec.y;
+
+        Classifier<Tuple>[] classifiers = new Classifier[k];
+        for (int i = 0; i < k; i++) {
+            int[][] yi = new int[n][1];
+            for (int j = 0; j < n; j++) {
+                yi[j][0] = y[j] == i ? pos : neg;
+            }
+
+            classifiers[i] = trainer.apply(formula, x.merge(DataFrame.of(yi, bv.name())));
+        }
+        return new OneVersusRest<>(classifiers);
+    }
+
     @Override
     public int predict(T x) {
         int y = 0;
         double maxf = Double.NEGATIVE_INFINITY;
         for (int i = 0; i < k; i++) {
-            double f = platts[i].scale(classifiers[i].f(x));
+            double f = platts != null? platts[i].scale(classifiers[i].f(x)) : classifiers[i].predict(x);
```

Comment:
here this part is trickier than I thought because I assumed classifiers[i].predict(x) would return a double that would be the predicted probability, when actually it returns a int that is either 0 or 1 in the one versus rest strategy. so here the predicted class is the first class that is positively predicted. for instance if we have 3 classes and we train 3 classifiers with the one versus rest strategy, say the first classifier predict probability of .6 for the first class, the second classifier predict probability of .3 for the second class, and the third classifier predict probability of .9 for the third class, the predicted class would be the first class here, when it should be the third one. 
In my tests, I still got 92% accuracy on usps dataset with one versus rest random forest (balanced, so I tricked the classWeight parameter to balance the classes by default), which mislead me to think this implementation was correct.